src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3780,7 +3780,9 @@ and genClause
 
     (genBar
      +> (fun ctx ->
-         if hasMultipleClausesWhereOneHasStroustrup then
+         let isMultiLineExpr = e.Range.StartLine <> e.Range.EndLine
```

Comment:
The input source code should never be used to verify whether something should multiline or not.
Fantomas doesn't use this anywhere and it will lead to hairy situations like


```fsharp
match x with
| Case1 -> [ "looooooooooooooooooooooooooooooooooooooooonnnnnnnnnnnnnngggggggggggggggggggggggggg"; "a";"b";"c" ]
// User feedback: Why isn't this transformed into multiple lines? It doesn't respect the max_line_length!
```

In both cases, the formatted result would be influenced by the original author of the code.
When used inside a team, the result will no longer be exactly the same regardless of the author.

I'll take a look what we could do instead to resolve this issue.